### PR TITLE
Create categories filter UI (dummy). #341

### PR DESCRIPTION
**Pull request by lemonboston**
This pull request contains the first form of the categories filter UI widget.

Outstanding issues, questions that may need to be addressed in separate tickets before this can be considered ready for a public release: https://github.com/schedjoules/general-issues/issues/197

Note, I may come back to update (and squash) this branch while I will be working on wiring in the list update, so please check, update each time.

### CODE REVIEW

**Comment by dmfs** [comment]
please rebase to master and resolve conflicts, I don't think we need categories/master anymore.

**Comment by lemonboston** [comment]
I've reset `categories/master` to `master` and rebased this one on top.

**Comment by dmfs** [comment]
For now I have one question: Instead of having these `CategoryOption` objects, wouldn't it be easier if you just have twp `Iterables` one for all available Categories and another one for the selected categories? The first one wouldn't have to change and the later one is probably easier to manage as well.

**Comment by lemonboston** [comment]
Two `Iterable`s would certainly be an option, but I like that `CategoryOption` fits well as the model for `FilterItemView`, so it get's updated with a category and whether that is selected or not.
And similarly `Iterable<CategoryOption>` fits well as the 'model' for the whole filter (there is private `update` method with that. With the 2 `Iterable`s we need to keep them in sync, if let's say we will handle language change.

File: eventdiscovery-sdk/src/main/res/layout/schedjoules_view_filter_item_clear.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout>
+
+    <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (dmfs) [inline eventdiscovery-sdk/src/main/res/layout/schedjoules_view_filter_item_clear.xml]: What's this FrameLayout for?

**Comment by lemonboston** [comment]
Replaced margins with paddings as discussed.

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/filter/categoryoption/NegateSelected.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import com.schedjoules.client.eventsdiscovery.Category;
+
+
+/**
+ * {@link CategoryOption} decorator negating {@link CategoryOption#isSelected()}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class NegateSelected implements CategoryOption
```
Comment (dmfs) [inline eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/filter/categoryoption/NegateSelected.java]: How about the name `SelectionComplement`?

File: eventdiscovery-sdk/src/main/res/layout/schedjoules_view_divider.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (dmfs) [inline eventdiscovery-sdk/src/main/res/layout/schedjoules_view_divider.xml]: What's the FrameLayout good for?

**Review by dmfs — CHANGES_REQUESTED** [review]
I do have a couple of concerns regarding the design. For the sake of progress I'll accept them since it seems to work nevertheless.

File: eventdiscovery-sdk/src/main/res/layout/schedjoules_view_filter_item_clear.xml (hunk 1 above)
Comment (lemonboston) [reply eventdiscovery-sdk/src/main/res/layout/schedjoules_view_filter_item_clear.xml]: Nothing, copied from the item layout and left in mistakenly. I removed it.

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/filter/categoryoption/NegateSelected.java (hunk 2 above)
Comment (lemonboston) [reply eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/filter/categoryoption/NegateSelected.java]: Renamed to `SelectionToggled` as suggested on chat.

File: eventdiscovery-sdk/src/main/res/layout/schedjoules_view_divider.xml (hunk 3 above)
Comment (lemonboston) [reply eventdiscovery-sdk/src/main/res/layout/schedjoules_view_divider.xml]: I remembered as if I had some problem with simply using a View as root, but now I tried and it works, so I changed it.

**Comment by lemonboston** [comment]
Ready for check again.
The design in `EventFilterView` could certainly be improved later, imo, too.

**Review by dmfs — APPROVED** [review]

**Pull request merged**